src/Disks/IO/CachedOnDiskWriteBufferFromFile.cpp

```diff
@@ -285,11 +284,17 @@ void CachedOnDiskWriteBufferFromFile::cacheData(char * data, size_t size)
             return;
         }
 
+        if (throw_on_error)
+            throw;
+
         tryLogCurrentException(__PRETTY_FUNCTION__);
         return;
```

Comment:
Shouldn't we re-throw `LOGICAL_ERROR` not depending on the setting? `cache_writer->write` has bunch of such checks.

Reply:
Hm, it could be done, it is already done for caching during read operations and not only for logical errors, but all errors from cache during SELECT query are rethrown. But here - for caching during writes - I did not do that, because caching during writes is a more young feature, so I was a bit afraid. Also in this PR I replace cache error codes so that all of them will be logical errors. I'd wait a bit more before enabling of rethrowing of all errors.